test/regress/cli/run_regression.py

```diff
@@ -174,11 +174,106 @@ def applies(self, benchmark_info):
     def run(self, benchmark_info):
         return super().run(
             benchmark_info._replace(
-                command_line_args=benchmark_info.command_line_args + ["--check-proofs", "--proof-granularity=theory-rewrite"]
+                command_line_args=benchmark_info.command_line_args +
+                ["--check-proofs", "--proof-granularity=theory-rewrite"]
             )
         )
 
 
+class LfscTester(Tester):
+    def applies(self, benchmark_info):
+        return (
+            benchmark_info.benchmark_ext != ".sy"
+            and (
+                "unsat" in benchmark_info.expected_output.split()
+                or "entailed" in benchmark_info.expected_output.split()
+            )
+            and "-i" not in benchmark_info.command_line_args
+            and "--incremental" not in benchmark_info.command_line_args
+            and "--no-produce-proofs" not in benchmark_info.command_line_args
+            and "--no-check-proofs" not in benchmark_info.command_line_args
+            and "--check-proofs" not in benchmark_info.command_line_args
+            and "--produce-unsat-cores" not in benchmark_info.command_line_args
+            and "--dump-unsat-cores" not in benchmark_info.command_line_args
+            and ":incremental true" not in benchmark_info.benchmark_content
+            and ":produce-proofs false" not in benchmark_info.benchmark_content
+            and ":check-proofs" not in benchmark_info.benchmark_content
+            and ":produce-unsat-cores true" not in benchmark_info.benchmark_content
+            and ":dump-unsat-cores true" not in benchmark_info.benchmark_content
+            and "(reset)" not in benchmark_info.benchmark_content
+        )
+
+    def run(self, benchmark_info):
+        exit_code = None
+        tmpf_name = None
+        with tempfile.NamedTemporaryFile(delete=False) as tmpf:
```

Comment:
We do delete the file once we're done with it (and there are no errors). If you set `delete` to `True`, the temporary file will be deleted once it's closed. However, we need it to stay where it is until we run `lfsc` on it. Only then can we delete. I think that's why you set it to `False` when you wrote this code for the `DumpTester` 😁.

Reply:
Hmm, I think that we should just move the code into the `with` block then. I am not quite sure what I was thinking when I wrote the code for the `DumpTester` :)

Reply:
OK. I moved everything inside the `with` block. Notice that this is [not portable](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile):
> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later)

Reply:
Hmm, I am wondering whether that was the reason for doing it differently? Let's keep it like this for now and see whether it causes problems even if the temporary file is opened by a different process (once we actually run regressions on Windows).